3rdparty/symcrypt_provider/CMakeLists.txt

```diff
@@ -0,0 +1,188 @@
+  ${SYMCRYPT_PROVIDER_DIR}/src/p_scossl_base.c
+  ${SYMCRYPT_COMMON_DIR}/src/scossl_helpers.c
+  ${SYMCRYPT_COMMON_DIR}/src/scossl_aes_aead.c
+  ${SYMCRYPT_COMMON_DIR}/src/scossl_dh.c
+  ${SYMCRYPT_COMMON_DIR}/src/scossl_ecc.c
+  ${SYMCRYPT_COMMON_DIR}/src/scossl_hkdf.c
+  ${SYMCRYPT_COMMON_DIR}/src/scossl_mac.c
+  ${SYMCRYPT_COMMON_DIR}/src/scossl_rsa.c
+  ${SYMCRYPT_COMMON_DIR}/src/scossl_sshkdf.c
+  ${SYMCRYPT_COMMON_DIR}/src/scossl_tls1prf.c)
+
+enclave_compile_definitions(oesymcryptprovider PRIVATE -D_AMD64_)
+
+if (uppercase_CMAKE_BUILD_TYPE STREQUAL "DEBUG")
+  enclave_compile_definitions(oesymcryptprovider PRIVATE -DDBG=1)
+endif ()
+
+enclave_include_directories(
+  oesymcryptprovider
+  PRIVATE
+  ${SYMCRYPT_PROVIDER_DIR}/inc
+  ${SYMCRYPT_PROVIDER_DIR}/src
+  ${SYMCRYPT_COMMON_DIR}/inc
+  ${SYMCRYPT_COMMON_DIR}/src
+  ${CMAKE_CURRENT_BINARY_DIR}/SymCrypt/inc
+  ${CMAKE_BINARY_DIR}/3rdparty/openssl/build/include_3 # for configuration.h
+  ${OPENSSL_3_DIR}/include)
+
+maybe_build_using_clangw(oesymcryptprovider)
+
+if (UNIX)
+  string(REPLACE "." ";" CLANG_VERSION_LIST ${CMAKE_C_COMPILER_VERSION})
+  list(GET CLANG_VERSION_LIST 0 CLANG_MAJOR_VERSION)
+
+  set(CLANG_INTRINSIC_HEADERS_SEARCH_PATH
+      "/usr/lib/llvm-${CLANG_MAJOR_VERSION}/lib/clang/${CMAKE_C_COMPILER_VERSION}/include"
+  )
+elseif (WIN32)
+  find_program(
+    CLANG_COMPILER clang
+    PATHS "C:/Program Files/LLVM/bin"
+    NO_DEFAULT_PATH REQUIRED)
+  get_filename_component(CLANG_COMPILER_ROOT_DIR ${CLANG_COMPILER} DIRECTORY)
+  file(GLOB CLANG_INTRINSIC_HEADERS_SEARCH_PATH
+       "${CLANG_COMPILER_ROOT_DIR}/../lib/clang/*/include")
+endif ()
+
+# Clang major version should match LLVM version
+find_path(
+  CLANG_INTRINSIC_HEADERS_DIR
+  NAMES "emmintrin.h"
+  PATHS ${CLANG_INTRINSIC_HEADERS_SEARCH_PATH}
+  NO_DEFAULT_PATH)
+
+if (CLANG_INTRINSIC_HEADERS_DIR)
+  if (WIN32)
+    # On Windows, the path includes spaces that the current clangw script cannot handle. As a workaround,
+    # we copy the directory to the CMAKE_CURRENT_BINARY_DIR/include
+    add_custom_target(
+      clang_intrinsic_headers_copy_provider
+      COMMAND ${CMAKE_COMMAND} -E copy_directory ${CLANG_INTRINSIC_HEADERS_DIR}
+              ${CMAKE_CURRENT_BINARY_DIR}/include)
+
+    add_enclave_dependencies(oesymcryptprovider
+                             clang_intrinsic_headers_copy_provider)
+
+    set(CLANG_INTRINSIC_HEADERS_DIR ${CMAKE_CURRENT_BINARY_DIR}/include)
+  endif ()
+
+  enclave_compile_options(oesymcryptprovider PRIVATE
+                          -I${CLANG_INTRINSIC_HEADERS_DIR})
+else ()
+  message(
+    FATAL_ERROR
+      "Cannot find clang intrinstics header (required by SymCrypt OpenSSL provider) in ${CLANG_INTRINSIC_HEADERS_SEARCH_PATH}"
+  )
+endif ()
+
+enclave_compile_options(
+  oesymcryptprovider
+  PRIVATE
+  # SymCrypt OpenSSL engine flags
+  # -maes
+  # -mpclmul
+  # -mrdrnd
+  # -mrdseed
+  # -mssse3
+  # -mxsave
+  # Supress warnings from SymCrypt OpenSSL engine
```

Comment:
Are these warning suppressions from compiling the engine also applicable to the provider?

Reply:
Please clarify - your comment here says "SymCrypt OpenSSL engine".